### ISSUE TITLE
Built in debugger

### ISSUE THREAD

**Issue by jerabaul29** [onivim/oni2#2437]
Any project / thoughts around having a debugger built in?

**Comment by bryphe** [comment]
Yes, I would really like to have this! I think a debugger with a keyboard-accessible-vim-style UI would be amazing.

**Comment by tosiek88** [comment]
Maybe it will be possible to go same road as in Emacs? I am using Doom Emacs with dap ( [debug adapter protocole](https://microsoft.github.io/debug-adapter-protocol/) ) -debug mode . For daily basis I am working with Typescript - React, NestJS, Node, and this debugger is totally fine - [gallery](https://emacs-lsp.github.io/dap-mode/screenshots/javascript.png).  For sure there will be still some problems to get working with C# due license issue.
